src/core_plugins/metrics/server/lib/vis_data/helpers/get_index_pattern.js

```diff
@@ -17,20 +17,25 @@
  * under the License.
  */
 
-import buildProcessorFunction from '../build_processor_function';
-import _ from 'lodash';
-import processors from '../response_processors/table';
+export async function getIndexPatternObject(req, indexPatternString) {
+  // getting the matching index pattern
+  const savedObjectClient = req.getSavedObjectsClient();
+  const indexPatternObjects = await savedObjectClient.find({
+    type: 'index-pattern',
+    fields: ['title', 'fields'],
+    search: `"${indexPatternString}"`,
+    search_fields: ['title'],
+  });
 
-export default function handleResponseBody(panel) {
-  return resp => {
-    if (resp.error) {
-      const err = new Error(resp.error.type);
-      err.response = JSON.stringify(resp);
-      throw err;
-    }
-    return panel.columns.map(column => {
-      const processor = buildProcessorFunction(processors, resp, panel, column);
-      return _.first(processor([]));
+  // getting the index pattern fields
+  const indexPatterns = indexPatternObjects.saved_objects
```

Comment:
There's already a utility for this here: https://github.com/elastic/kibana/blob/837a0544669484a2af052adf7dc03b5f0bd6899e/src/ui/public/index_patterns/static_utils/index.js#L28